contracts/AedexV2Factory.aes

```diff
@@ -39,8 +39,6 @@ contract AedexV2Factory =
             , allow_debug_mode = Option.default(false,allow_debug_mode)
             }
 
-    //TODO: this is a hack to store it's own address into state as a IAedexV2Factory
-    //try finding something more suitable
     stateful entrypoint set_this_factory( this_factory: IAedexV2FactoryForPair ) =
         require(Contract.address == this_factory.address,"AedexV2Factory: NOT_SAME_FACTORY")
         put( state { this_factory = Some(this_factory) })
```

Comment:
are we happy with this? if so I am fine with removing this todo but I can't find a paper trail of the discussion?

Reply:
Didn't find any other solution to pass the own address casted to some interface to a remote call. Indeed I was asking on the WIRE, but here is a good place to talk about it.

Reply:
So we can not set it during init and Contract.address is not available where we need it? Where would that be?

Reply:
I just saw its only the cloning. Contract.address is not an option there? I remeber we had this discussion already but i forgot the details...

Reply:
If I would be able to cast `Contract.address` to  `IAedexV2FactoryForPair` that would be great, but the Contract.address is of type `address`, so compiler complains if I pass it like that. 
The only option to fool the compiler was to receive it through a regular entrypoint as a `IAedexV2FactoryForPair` parameter and from 'ouside world' to give the address of the actual factory.

Reply:
what was the response of the core devs? did we raise that topic there?

Reply:
I was asking on the WIRE not on slack, nobody responded me with a alternative for this.

Reply:
then we should probably tag @hanssv here and ask him =)

Reply:
actually i found [this](https://github.com/aeternity/aesophia/blob/master/docs/sophia_stdlib.md#to_contract), i will try it and come back with news

Reply:
done it, made the modifications